### PR TITLE
fix(types): export Octokit type

### LINKED ISSUE #1624: Upgrading to v17 breaks TypeScript types
Hello! 

Here is the error:

```
'Octokit' refers to a value, but is being used as a type here.
```

```ts
import { Octokit } from "@octokit/rest";

class MyClass {             
  constructor(public logger: Logger, public octokit: Octokit) {}
                                                       ^--- error is here
}
```

I think the typings might just be pointing to the wrong thing somehow? I notice in `package.json` that [`types` refers to `index.d.ts`](https://github.com/octokit/rest.js/blob/master/package.json#L65) but when importing the package it is grabbing the types from `node_modules/@octokit/rest/dist-types/index.d.ts`.

When I open that file, this is all it contains:

```ts
import { Octokit as Core } from "@octokit/core";
export declare const Octokit: (new (...args: any[]) => {
    [x: string]: any;
}) & {
    new (...args: any[]): {
        [x: string]: any;
    };
    plugins: any[];
} & typeof Core & import("@octokit/core/dist-types/types").Constructor<void> & import("@octokit/core/dist-types/types").Constructor<{
    paginate: import("@octokit/plugin-paginate-rest").PaginateInterface;
} & import("@octokit/plugin-rest-endpoint-methods/dist-types/generated/types").RestEndpointMethods>;
```

TypeScript: 3.7.5
Node: 12.15
Yarn: 1.22.0

--- 

This code worked prior to v17 and I don't see anything in the changelog regarding changes to types.

Thanks!

### PR DESCRIPTION
**Pull request by jamescdavis**
closes #1624

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -17,3 +17,5 @@ export const Octokit = Core.plugin([
 ]).defaults({
   userAgent: `octokit-rest.js/${VERSION}`
 });
+
+export type Octokit = InstanceType<typeof Octokit>
```
Comment (gr2m) [inline src/index.ts]: could you add a test that fails without this addition, so we don't break that behavior in future?

Comment (jamescdavis) [reply src/index.ts]: It looks like https://github.com/octokit/rest.js/blob/master/test/typescript-validate.ts is no longer checked?

Comment (jamescdavis) [reply src/index.ts]: Oh, I see that was part of the TS rewrite. I'll add it to a jest test.

**Comment by jamescdavis** [comment]
@gr2m ok, I've added a failing test for using `Octokit` as a type (in `smoke.test.ts`, let me know if it should go somewhere else), let that run and fail, and then re-added the fix for green CI.

**Review by rwjblue — COMMENTED** [review]
Thank you!

**Review by gr2m — APPROVED** [review]
Wonderful work, thank you so much!

**Pull request merged**